### PR TITLE
[CI] Add new release note step for release 

### PR DESCRIPTION
**Pull request by ThibaudAV**
## Issue

https://gravitee.atlassian.net/browse/APIM-367

## Description

The goal is to remove the use of GitHub milestones 

This step open a PR into https://github.com/gravitee-io/gravitee-docs with empty Release Notes template to be completed.
And try to add some info to help to complete the Release Notes in PR body like commit list and link to Jira filtered to "Done" issues of this release 

Pr linked : https://github.com/gravitee-io/gravitee-docs/pull/912

## Additional context

Pr example : https://github.com/gravitee-io/gravitee-docs/pull/910

📝 BackPorted on each supported version when(if) this PR are ✅ 

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.z6.web.core.windows.net/ci-changelog/index.html)
_Notes_: The deployed app is linked to the management API of APIM master. (Same login and password as APIM master)

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-dxpcmrdixk.chromatic.com)

### CODE REVIEW

File: release/steps/4-generate_changelog.mjs [hunk 1]
```diff
@@ -10,33 +10,17 @@ const releasingVersion = await extractVersion();
 
 console.log(chalk.blue(`Triggering Changelog generation Pipeline`));
 
-const milestone = `APIM - ${releasingVersion}`;
-
 // Use the preconfigured payload from config folder with the good parameters
 // For changelog generation, always work on master
 const body = {
-  branch: 'master',
```
Comment (ThibaudAV) [inline release/steps/4-generate_changelog.mjs:18]: WIP to keep 'master'

File: release/ci-steps/doc-new-release-notes.mjs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+echo(chalk.blue(`# Get feat & fix commits`));
+const allFeatCommits = await $`git log $(git describe --tags --abbrev=0)..HEAD --no-merges --oneline --grep "^feat\\|^perf"`;
+const allFixCommits = await $`git log $(git describe --tags --abbrev=0)..HEAD --no-merges --oneline --grep "^fix"`;
+
+echo(chalk.blue(`# Clone gravitee-docs repository`));
+await $`mkdir -p release-notes`;
+cd('release-notes');
+await $`rm -rf gravitee-docs`;
+await $`git clone --depth 1  ${docRepository} --single-branch --branch=apim-release-notes`;
```
Comment (ThibaudAV) [inline release/ci-steps/doc-new-release-notes.mjs]: to change to master

File: release/ci-steps/doc-new-release-notes.mjs [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+console.log(chalk.magenta(`#############################################`));
+
+const releasingVersion = await extractVersion();
+const versions = computeVersion(releasingVersion);
+
+const docRepository = 'https://github.com/gravitee-io/gravitee-docs';
+const docApimReleaseNotesFolder = 'pages/apim/3.x/changelog/';
+const docApimReleaseNotesFile = `${docApimReleaseNotesFolder}changelog-${versions.trimmed}.adoc`;
+
+const gitBranch = `release-${releasingVersion}`;
```
Comment (phiz71) [inline release/ci-steps/doc-new-release-notes.mjs]:
```suggestion
const gitBranch = `release-apim-${releasingVersion}`;
```

File: release/ci-steps/doc-new-release-notes.mjs [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+${allFixCommits.stdout}
+</details>
+
+## Jira issues
+
+[See all Jira issues for ${versions.branch} version](https://gravitee.atlassian.net/jira/software/c/projects/APIM/issues/?jql=project%20%3D%20%22APIM%22%20and%20fixVersion%20%3D%20${versions.branch}%20and%20status%20%3D%20Done%20ORDER%20BY%20created%20DESC)
+`;
+echo(chalk.blue('# Create PR on Github Doc repository'));
+echo(prBody);
```
Comment (phiz71) [inline release/ci-steps/doc-new-release-notes.mjs:140]: Maybe not necessary to print the description of the PR. We will see it in the PR.

File: release/steps/4-generate_release_notes.mjs [hunk 5]
```diff
@@ -8,35 +8,19 @@ await checkToken();
...
-
-const milestone = `APIM - ${releasingVersion}`;
+console.log(chalk.blue(`Triggering Release Notes generation Pipeline`));
 
 // Use the preconfigured payload from config folder with the good parameters
-// For changelog generation, always work on master
+// For release-notes generation, always work on master
 const body = {
-  branch: 'master',
+  branch: 'ci-changelog',
```
Comment (phiz71) [inline release/steps/4-generate_release_notes.mjs]:
```suggestion
  branch: 'master',
```

**Review by phiz71 — CHANGES_REQUESTED** [review]

**Comment by phiz71** [comment]
Why is there a submodule created ?
image

File: release/ci-steps/doc-new-release-notes.mjs (hunk 3 above)
Comment (ThibaudAV) [reply release/ci-steps/doc-new-release-notes.mjs]: +1

File: release/steps/4-generate_release_notes.mjs (hunk 5 above)
Comment (ThibaudAV) [reply release/steps/4-generate_release_notes.mjs]: It's for testing I'll change them at the last moment. see my 2 comments above ;)

**Comment by ThibaudAV** [comment]
> Why is there a submodule created ?
oups to remove

File: release/ci-steps/doc-new-release-notes.mjs (hunk 4 above)
Comment (phiz71) [reply release/ci-steps/doc-new-release-notes.mjs:140]: after discussion, you can leave it for debug purpose

**Review by phiz71 — APPROVED** [review]

**Pull request merged**